current/runtime/fees.md

```diff
@@ -4,118 +4,151 @@ lang: en
+  [`TransactionByteFee`](https://substrate.dev/rustdocs/master/pallet_transaction_payment/trait.Trait.html#associatedtype.TransactionByteFee).
+* `weight_fee`: A per-weight-unit fee that is multiplied by the weight of the transaction. The
+  weight of each dispatch is denoted via the flexible `#[weight]` annotation. The weight must be
+  converted to the `Currency` type. For this, each runtime must define a
+  [`WeightToFee`](https://substrate.dev/rustdocs/master/pallet_transaction_payment/trait.Trait.html#associatedtype.WeightToFee)
+  type that makes the conversion. `WeightToFee` must be a struct that implements [`Convert<Weight,
+  Balance>`](https://substrate.dev/rustdocs/master/sp_runtime/traits/trait.Convert.html).
+
+The sum of the three transaction fee components (i.e. `base_fee + length_fee + weight_fee`) is known
+as the _inclusion fee_. Even if a transaction fails, the signer must pay the inclusion fee.
+
+## Default Weight Fees
+
+The default fee system uses _fixed_ weights, which means that the arguments to a dispatch do not
+affect its weight. The tiers in this system are represented by the following enum:
 
 ```rust
 pub enum SimpleDispatchInfo {
-    /// A normal dispatch with fixed weight.
     FixedNormal(Weight),
-    /// A normal dispatch with the maximum weight.
     MaxNormal,
-    /// A normal dispatch with no weight.
-    FreeNormal,
-    /// An operational dispatch with fixed weight.
+    InsecureFreeNormal,
     FixedOperational(Weight),
-    /// An operational dispatch with the maximum weight.
     MaxOperational,
-    /// An operational dispatch with no weight.
-    FreeOperational,
+    FixedMandatory(Weight),
 }
 ```
 
-This enum groups all dispatches into _normal_ or _operational_ and gives them a fixed weight. This
-means that the arguments of a dispatchable function do not affect the weight. A dispatch classified
-as _operational_ is exempt from paying both the `base_fee` and the `length_fee`.
-
-### Free Normal
-
-This means that this function has no weight. It will not contribute to block fullness at all, and no
-weight-fee is applied.
+In order to delineate their purposes, the enums in this group are separated into _dispatch classes_,
+which are also defined by an enum:
 
 ```rust
-#[weight = SimpleDispatchInfo::FreeNormal]
-pub fn some_normal_function_light() { noop(); }
+pub enum DispatchClass {
+    Normal,
+    Operational,
+    Mandatory,
+}
 ```
 
-### Fixed Normal
+### Normal Dispatches
 
-This function will always have a weight `10`.
+Dispatches in this class represent normal user-triggered transactions. These types of dispatches may
+only consume a portion of a block's total weight limit; this portion can be found by examining the
+[`AvailableBlockRatio`](https://substrate.dev/rustdocs/master/frame_system/trait.Trait.html#associatedtype.AvailableBlockRatio).
+Normal dispatches are assigned a priority that is proportional to their weight.
```

Comment:
This might be true but probably belongs in the tx queue document, which should handle priority of all extrinsics. You could say that normal dispatches go into the tx queue and given priority based on the queue's logic.